src/main/java/no/unit/nva/bare/BareConnection.java

```diff
@@ -78,63 +73,75 @@ protected URI generateGetUrl(String systemControlNumber)
-                throw new IOException(response.getStatusLine().getReasonPhrase());
-            }
+        HttpRequest request = HttpRequest.newBuilder()
+                .GET()
+                .version(HttpClient.Version.HTTP_2)
+                .uri(getUrl)
+                .timeout(TIMEOUT_DURATION)
+                .build();
+        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        if (response.statusCode() == Response.Status.OK.getStatusCode()) {
+            final String body = response.body();
+            return new Gson().fromJson(body, BareAuthority.class);
+        } else {
+            System.out.println("Error..? " + response.body());
+            throw new IOException(response.body());
         }
     }
 
     /**
      * Updates metadata of the given authority to Bare.
      *
      * @param authoritySystemControlNumber Identifier of Authority to update
-     * @param authorityIdentifier New identifier pair to add to authority
+     * @param authorityIdentifier          New identifier pair to add to authority
      * @return CloseableHttpResponse
      * @throws IOException        communication error
      * @throws URISyntaxException error while creating URI
+     * @throws InterruptedException error in communication
      */
-    public CloseableHttpResponse addIdentifier(String authoritySystemControlNumber,
-                                               AuthorityIdentifier authorityIdentifier) throws IOException,
-            URISyntaxException {
+    public HttpResponse<String> addIdentifier(String authoritySystemControlNumber,
+                                              AuthorityIdentifier authorityIdentifier) throws IOException,
+            URISyntaxException, InterruptedException {
         URI uri = new URIBuilder()
                 .setScheme(HTTPS)
                 .setHost(Config.getInstance().getBareHost())
                 .setPathSegments(PATH_SEGMENT_AUTHORITY, PATH_SEGMENT_REST, PATH_SEGMENT_AUTHORITIES, PATH_SEGMENT_V_2,
                         authoritySystemControlNumber, PATH_SEGMENT_IDENTIFIERS)
                 .build();
         System.out.println("uri=" + uri);
-        HttpPost httpPost = new HttpPost(uri);
 
+        final String body = new Gson().toJson(authorityIdentifier, AuthorityIdentifier.class);
+        HttpRequest.BodyPublisher bodyPublisher = HttpRequest.BodyPublishers.ofString(body);
         String apiKeyAuth = APIKEY_KEY + SPACE + Config.getInstance().getBareApikey();
-        httpPost.addHeader("Authorization", apiKeyAuth);
-        httpPost.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
-        httpPost.setEntity(new StringEntity(new Gson().toJson(authorityIdentifier, AuthorityIdentifier.class)));
-        System.out.println("httpPost=" + httpPost);
-        return httpClient.execute(httpPost);
+
+        HttpRequest request = HttpRequest.newBuilder()
```

Comment:
Possibly extract as method for reuse below

Reply:
Acctually I do not see how we could gain here?!
Maybe if we take only the requestBuilder in a extra method. But we sitt have to do the POST(), GET(), etc. pp. in the method itself:

`final HttpRequest.Builder requestBuilder = getHttpRequestBuilder(uri);
        HttpRequest request = requestBuilder.POST(bodyPublisher).build();`